e2e/pages/volumes-page.ts

```diff
@@ -0,0 +1,140 @@
+import type {Locator, Page} from '@playwright/test';
+
+export class VolumesPage {
+  readonly page: Page;
+  readonly table: Locator;
+  readonly namespaceSelector: Locator;
+  readonly searchBox: Locator;
+  readonly errorBanner: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.table = page.locator('.volumesTable');
+    this.namespaceSelector = page.locator('.select-namespace');
+    this.searchBox = page.locator('.search-box input');
+    this.errorBanner = page.locator('.banner.error');
+  }
+
+  getVolumeRow(volumeName: string) {
+    return this.page.locator('tr.main-row').filter({hasText: volumeName});
+  }
+
+  async waitForVolumeToAppear(volumeName: string, timeout = 30000) {
+    const volumeRow = this.getVolumeRow(volumeName);
+    await volumeRow.waitFor({state: 'visible', timeout});
```

Comment:
```suggestion
    await expect(volumeRow).toBeVisible();
```
(Or do it in the caller.)